packages/components/data-table-manager/src/data-table-manager.stories.tsx

```diff
@@ -177,8 +215,66 @@ export const BasicExample: Story = (args) => {
     visibleColumnKeys: initialVisibleColumns.map(({ key }) => key),
   });
 
+  const [customTableData, setCustomTableData] = useState({
+    columns: initialColumnsState,
+    visibleColumnKeys: initialVisibleColumns.map(({ key }) => key),
+  });
+
+  const customColumnManager = {
+    areHiddenCustomColumnsSearchable: true,
+    // @ts-expect-error
+    searchHiddenColumns: (searchTerm) => {
+      setCustomTableData({
+        ...customTableData,
+        columns: initialColumnsState.filter(
+          (column) =>
+            customTableData.visibleColumnKeys.includes(column.key) ||
+            column.label
+              .toLocaleLowerCase()
+              .includes(searchTerm.toLocaleLowerCase())
+        ),
+      });
+    },
+    disableCustomColumnManager: false,
+    visibleColumnKeys: customTableData.visibleColumnKeys,
+    hideableColumns: customTableData.columns,
+  };
+
+  const mappedCustomColumns = customColumnManager.hideableColumns.reduce(
+    (columns, column) => ({
+      ...columns,
+      [column.key]: column,
+    }),
+    {}
+  );
+
   const [isCondensed, setIsCondensed] = useState(true);
   const [isWrappingText, setIsWrappingText] = useState(false);
+  const [rowClick, setRowClick] = useState(true);
+  const [rowSelection, setRowSelection] = useState(true);
+  const [textColor, setTextColor] = useState('black');
+  const [disableResize, setDisableResize] = useState(true);
+
+  const updateRowSelection = (
+    newState: boolean | ((prevState: boolean) => boolean)
+  ) => {
+    setRowSelection(newState);
+  };
+  const updateRowClick = (
+    newState: boolean | ((prevState: boolean) => boolean)
+  ) => {
+    setRowClick(newState);
+  };
+  const updateTextColor = (
+    newState: string | ((prevState: string) => string)
+  ) => {
+    setTextColor(newState);
+  };
+  const updateColumnResize = (
+    newState: boolean | ((prevState: boolean) => boolean)
+  ) => {
+    setDisableResize(newState);
+  };
```

Comment:
Anything speaking against using the `setState` functions directly instead of proxying them? 

Oh, and you can define the expected type of accepted values right at the hook (if a payload is in the wrong shape your editor will let you know).

```
const [rowClick, setRowClick] = useState<boolean>(true);
```

Reply:
> Oh, and you can define the expected type of accepted values right at the hook (if a payload is in the wrong shape your editor will let you know).

Thank you, updated here: https://github.com/commercetools/ui-kit/pull/2938/commits/d4ad05447ab56bad68d7fac6104c085a9943877e

> Anything speaking against using the setState functions directly instead of proxying them?

This is like so because we are actually managing the state from a parent component(in this context, where it is global enough for other components to reference the same state for the data table settings) and then the state it triggered from a child component or any custom component that is created by the user. For example: 

https://github.com/commercetools/ui-kit/blob/ccf69ac02cc7aa74e484a07d5413a942d74e8975/packages/components/data-table-manager/src/data-table-manager.stories.tsx#L1022